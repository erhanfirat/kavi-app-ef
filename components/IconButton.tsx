import { StyleSheet, Text, TouchableOpacity } from "react-native";
import { IconSymbol } from "./ui/IconSymbol";

type IconButtonProps = {
  title?: string;
  icon?: import("expo-symbols").SymbolViewProps["name"];
  onPress: () => void;
};

const IconButton: React.FC<IconButtonProps> = ({ icon, title, onPress }) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      {icon && <IconSymbol size={28} name={icon} color={styles.text.color} />}
      {title && <Text style={styles.text}>{title}</Text>}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#007BFF",
    padding: 10,
    borderRadius: 5,
  },
  text: {
    color: "#fff",
    marginLeft: 5,
    fontSize: 16,
  },
});

export default IconButton;
