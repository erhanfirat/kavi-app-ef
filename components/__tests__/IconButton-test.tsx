import * as React from "react";
import renderer from "react-test-renderer";
import IconButton from "../IconButton";

it(`Icon Button renders correctly`, () => {
  const tree = renderer
    .create(
      <IconButton
        title="search"
        icon="magnifyingglass"
        onPress={() => alert("merhaba!")}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
