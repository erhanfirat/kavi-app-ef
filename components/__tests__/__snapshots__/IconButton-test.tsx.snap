// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#007BFF",
      "borderRadius": 5,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 10,
    }
  }
/>
`;
